### PR TITLE
ADEN-5263 Adjust regular ad units to remove some of the variance

### PR DESCRIPTION
**Pull request by v3rth**

### CODE REVIEW

File: extensions/wikia/AdEngine/js/provider/directGpt.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
 	'ext.wikia.adEngine.context.uapContext',
 	'ext.wikia.adEngine.provider.factory.wikiaGpt',
+	'ext.wikia.adEngine.slot.adUnitBuilder',
+	'ext.wikia.adEngine.slot.service.kiloAdUnitBuilder',
 	'ext.wikia.adEngine.slotTweaker',
 	require.optional('ext.wikia.adEngine.lookup.openx.openXBidderHelper'),
 	require.optional('ext.wikia.aRecoveryEngine.pageFair.recovery'),
 	require.optional('ext.wikia.aRecoveryEngine.sourcePoint.recovery')
-], function (uapContext, factory, slotTweaker, openXHelper, pageFair, sourcePoint) {
+], function (adContext, uapContext, factory, adUnitBuilder, kiloAdUnitBuilder, slotTweaker, openXHelper, pageFair, sourcePoint) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/provider/directGpt.js:13]: MAJOR Function has 9 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/spec/provider/gptProviders.spec.js [hunk 2]
```diff
@@ -27,6 +27,9 @@ describe('ext.wikia.adEngine.provider.*', function () {
 			},
 			buildNew: noop
 		},
+		kiloAdUnitBuilder: {
+			build: function () {}
+		},
 		gptHelper: {
 			pushAd: function (slotName, slotElement, slotPath, slotTargeting, extra) {
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/spec/provider/gptProviders.spec.js:34]: MAJOR Function has 5 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 9 major

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [adContext.js#L14](https://github.com/Wikia/app/blob/05a80666f9d0315b12839b1b94e0782e560505e3/extensions/wikia/AdEngine/js/context/adContext.js#L14): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)
1. ![MAJOR][MAJOR] [adContext.js#L74](https://github.com/Wikia/app/blob/05a80666f9d0315b12839b1b94e0782e560505e3/extensions/wikia/AdEngine/js/context/adContext.js#L74): Function has a complexity of 19 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [adContext.js#L96](https://github.com/Wikia/app/blob/05a80666f9d0315b12839b1b94e0782e560505e3/extensions/wikia/AdEngine/js/context/adContext.js#L96): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [adContext.js#L131](https://github.com/Wikia/app/blob/05a80666f9d0315b12839b1b94e0782e560505e3/extensions/wikia/AdEngine/js/context/adContext.js#L131): Function has a complexity of 31 which is greater than 10 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
1. ![MAJOR][MAJOR] [adContext.js#L207](https://github.com/Wikia/app/blob/05a80666f9d0315b12839b1b94e0782e560505e3/extensions/wikia/AdEngine/js/context/adContext.js#L207): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1067)
1. ![MAJOR][MAJOR] [adContext.js#L239](https://github.com/Wikia/app/blob/05a80666f9d0315b12839b1b94e0782e560505e3/extensions/wikia/AdEngine/js/context/adContext.js#L239): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)
1. ![MAJOR][MAJOR] [AdContext.spec.js](https://github.com/Wikia/app/blob/05a80666f9d0315b12839b1b94e0782e560505e3/extensions/wikia/AdEngine/js/spec/AdContext.spec.js): File "AdContext.spec.js" has 1012 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS104)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues
* ![MAJOR][MAJOR] 20 major
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$title". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$article". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$output". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$user". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L22](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L22): Remove the unused function parameter "$wiki". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L100](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L100): Remove the unused function parameter "$scripts". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L101](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L101): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L197](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L197): Remove the unused function parameter "$skin". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L222](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L222): Remove the unused function parameter "$jsExtensionPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)
1. ![MAJOR][MAJOR] [AdEngine2Hooks.class.php#L222](https://github.com/Wikia/app/blob/40ab5109031b322375089afb1bc1d8256f35593f/extensions/wikia/AdEngine/AdEngine2Hooks.class.php#L222): Remove the unused function parameter "$scssPackages". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1172)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by nandy-andy — APPROVED** [review]

**Pull request merged**